rules/TYPO312/v0/RemoveAddLLrefForTCAdescrMethodCallRector.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Ssch\TYPO3Rector\TYPO312\v0;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\StaticCall;
+use PHPStan\Type\ObjectType;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\Contract\DocumentedRuleInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use TYPO3\CMS\Core\Utility\ExtensionManagementUtility;
+
+/**
+ * @changelog https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/12.0/Deprecation-97312-DeprecateCSH-relatedMethods.html
+ * @see \Ssch\TYPO3Rector\Tests\Rector\v12\v0\RemoveAddLLrefForTCAdescrMethodCallRector\RemoveAddLLrefForTCAdescrMethodCallRectorTest
+ */
+final class RemoveAddLLrefForTCAdescrMethodCallRector extends AbstractRector implements DocumentedRuleInterface
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Remove ExtensionManagementUtility::addLLrefForTCAdescr() method call as it will be removed in TYPO3 v13',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+use TYPO3\CMS\Core\Utility\ExtensionManagementUtility;
+ExtensionManagementUtility::addLLrefForTCAdescr('_MOD_web_info', 'EXT:info/Resources/Private/Language/locallang_csh_web_info.xlf');
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+use TYPO3\CMS\Core\Utility\ExtensionManagementUtility;
+CODE_SAMPLE
+                ),
+
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [StaticCall::class];
+    }
+
+    /**
+     * @param StaticCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $node instanceof StaticCall) {
```

Comment:
this check is useless as this will always be the case because of the nodeTypes filter above